cleverhans/model.py

```diff
@@ -11,9 +11,18 @@ class Model(object):
     model when a specific package does not directly expose them.
     """
     __metaclass__ = ABCMeta
+    O_LOGITS, O_PROBS, O_FEATURES = 'logits probs features'.split()
```

Comment:
Could make these variables modifiable through some arguments in the `__init__` method? Otherwise, what is the purpose of adding these variables?

Reply:
The purpose of these variables is not to type strings when writing models and as such to avoid typo-related bugs. In addition, you get auto-complete benefits and it's self-documented (you don't have to enumerate valid strings in the doc).

Reply:
What does the O_ mean?

Reply:
It stands for output, I tend to prefix enums by a letter or a word so as to not confuse which function they are compatible with. I'm open to suggestions.